test/unit-test/core_mqtt_serializer_utest.c

```diff
@@ -1401,6 +1401,13 @@ void test_MQTT_DeserializeAck_connack( void )
     buffer[ 1 ] = 1;
     status = MQTT_DeserializeAck( &mqttPacketInfo, &packetIdentifier, &sessionPresent );
     TEST_ASSERT_EQUAL_INT( MQTTBadResponse, status );
+    TEST_ASSERT_EQUAL_INT( true, sessionPresent );
+
+    /* Session present not set but nonzero return code. */
+    buffer[ 0 ] = MQTT_PACKET_CONNACK_SESSION_PRESENT_MASK;
+    buffer[ 1 ] = 1;
+    status = MQTT_DeserializeAck( &mqttPacketInfo, &packetIdentifier, &sessionPresent );
+    TEST_ASSERT_EQUAL_INT( MQTTBadResponse, status );
```

Comment:
This is identical to the test case above it

Reply:
Fixed